src/Plivo/Resource/ResourceInterface.cs

```diff
@@ -175,8 +175,12 @@ public async Task<T> DeleteResource<T> (string id, Dictionary<string, object> da
                 return func ();
             } catch (AggregateException ex) {
                 ex.Flatten ();
+                if (ex.InnerExceptions[0] is Newtonsoft.Json.JsonReaderException){
```

Comment:
cant we check the PlivoAuthenticationException here?

Reply:
those exceptions are being handled at https://github.com/plivo/plivo-dotnet/blob/6426ae5217010841ca1d6ccad2b5ba1b4ef0af12/src/Plivo/Http/PlivoResponse.cs#L90

Reply:
Aggregate exceptions will catch all the exceptions including all defined Plivo.Exception.

Reply:
@abrolnalin Please confirm if `PlivoAuthenticationException` is raised when status code received is 401, regardless of whether it's json response in body or not.